### PR TITLE
Rates statistic

### PR DESCRIPTION
**Pull request by maroun2**
## Description
It adds a plugin with rates statistic. It seems that someone wants this future also #312
It's my first pull request, so please excuse my mistakes.
Preview: https://ibb.co/TPKM8ZJ

## TESTING STAGE
It was tested. This functionality is separated from lending functionality. 

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [ ] **I have read CONTRIBUTING.md**
- [x] **I fully understand [Github Flow.](https://guides.github.com/introduction/flow/)**
- [x] **My code adheres to the [code style of this project.](https://poloniexlendingbot.readthedocs.io/en/latest/contributing.html)**
- [ ] **I have updated the documentation in /docs if I have changed the config, arguments, logic in how the bot works, or anything that understandably needs a documentation change.**
- [x] **I have updated the config file accordingly if my change requires a new configuration setting or changes an existing one.**
- [x] **I have tested the bot with no issues for 24 continuous hours. If issues were experienced, they have been patched and tested again.**

### CODE REVIEW

**Comment by Evanito** [comment]
Currently: chart is based on the 24 hour schedule with multiple days stacked over each other.

This is useful if you are trying to look at the rates at certain times of day, otherwise I think it would be better to see the days linked end to end so you can see historical rates.

Maybe include a toggle between the modes or just set the default to the more intuitive mode (historical.)

File: plugins/Rates.py [hunk 1]
```diff
@@ -0,0 +1,82 @@
...
+            loan_order = self.api.return_loan_orders(currency, 10000)
+
+            btc_offer = 0
+            gap_botom_offer = 0
+            gap_top_offer = 0
+
+            offer_sum = 0
+            for offer in loan_order['offers']:
+                offer_sum += float(offer['amount'])
+                if btc_offer > 1 and gap_botom_offer == 0:
```
Comment (hajdbo) [inline plugins/Rates.py]: btc_offer is never > 0

File: plugins/Rates.py [hunk 2]
```diff
@@ -0,0 +1,82 @@
...
+            if os.path.isfile(self.rates_file):
+                with open(self.rates_file, "r") as file:
+                    file_data = json.loads(file.read())
+
+            data = {}
+            data[currency] = []
+            d = datetime.datetime.utcnow()
+            epoch = datetime.datetime(1970, 1, 1)
+            t = (d - epoch).total_seconds()
+            for row in file_data[currency]:
```
Comment (SloRunner) [inline plugins/Rates.py:71]: fails if there is no file, fixable by commenting and un commenting, so maybe add
`if len(file_data[currency]) > 0:`

**Review by SloRunner — CHANGES_REQUESTED** [review]

File: plugins/Rates.py (hunk 2 above)
Comment (SloRunner) [reply plugins/Rates.py:71]:
```suggestion
            if file_data:
              for row in file_data[currency]:
```

Tested and works as expected, else you get KeyError

**Pull request closed without merging**